clouddq/main.py

```diff
@@ -270,8 +278,26 @@ def main(  # noqa: C901
             gcp_service_account_key_path=gcp_service_account_key_path,
             gcp_impersonation_credentials=gcp_impersonation_credentials,
         )
+        # Load metadata
+        metadata = json.loads(metadata)
         # Set-up cloud logging
-        add_cloud_logging_handler(logger=json_logger)
+        if add_metadata_to_logs:
+            dataplex_batch_name = os.environ.get("DATAPLEX_BATCH_NAME")
+            dataplex_task_name = os.environ.get("DATAPLEX_TASK_NAME")
+            if dataplex_batch_name:
+                metadata["dataplex_batch_name"] = dataplex_batch_name
+            if dataplex_task_name:
+                metadata["dataplex_task_name"] = dataplex_task_name
+            add_cloud_logging_handler(logger=json_logger, metadata=metadata)
+        else:
+            dataplex_batch_name = os.environ.get("DATAPLEX_BATCH_NAME")
+            dataplex_task_name = os.environ.get("DATAPLEX_TASK_NAME")
+            if dataplex_batch_name:
+                metadata["dataplex_batch_name"] = dataplex_batch_name
+            if dataplex_task_name:
+                metadata["dataplex_task_name"] = dataplex_task_name
+            add_cloud_logging_handler(logger=json_logger, metadata=metadata)
```

Comment:
if not add_metadata_to_logs: should we not exclude the "metadata" dictionary and only include the dataplex_batch_name and dataplex_task_name?